### DISCUSSION TITLE
Notes with big code blocks are slow

### DISCUSSION THREAD

**Discussion by halyph** [glushchenko/fsnotes#1440] [Q&A] [answered]
Hi all,
I've just noticed that notes with several big code blocks have a big impact on FSNotes performance.
I mean FSNotes has significant freeze (~0.1-0.3 s) and slower scrolling.

The easiest way to reproduce this is to create one huge code block and start creating the 2nd via typing "\`". You will notice delay.

Do you have any advice how I can properly measure this and create easy to reproduce bug report? 
Maybe FSNotes has some back-doors to switch on some magic logging etc.

**Comment by glushchenko** [comment, marked as answer]
I know about. Just accept that FSNotes is not for large files. Any editor can be stacked with a long block of code, the question is long.

**Reply by halyph** [reply]
@glushchenko 
> the question is long.

do you mean I wrote very long question? Or the problem/solution is long to explain in one/two sentences?

**Reply by glushchenko** [reply]
Code highlighting is slow because it works through https://highlightjs.org/ JS library, do you want to make a port to Swift? 

In addition, the nstextview component itself is not very fast.

What's the question?

**Reply by halyph** [reply]
1. I don't have any other questions except what was stated above. I just didn't understand the initial reply. 
2. Also I couldn't write a port because I need 1st learn Swift etc.
3. I've been using Github Discussion because I didn't know how properly report this issue. And based on this discussion it looks like well known issue. I.e. it is not required to be reported.
4. I've been reporting other bugs. But based on the ton of discussion I am not sure whether you as maintainer need them or not. 

Feel free to remove them. Good luck.

**Reply by glushchenko** [reply]
There is no need to make a bug report about slow performance, I know a million ways to break the performance of the program.

In this case, low performance on these components is normal.

Again, the program is not suitable for large blocks of text, and I'm not going to do anything about that right now.

I am grateful for every bug report, if it is indeed a bug.
